test/unit/grpcServer/handlers/core/updateStateHandlerFactory.js

```diff
@@ -0,0 +1,178 @@
+        'B762539A7C17C33A65C46727BFCF2C701390E6AD7DE5190B6CC1CF843CA7E262',
+      height: '24',
+    };
+
+    rpcClientMock = {
+      request: this.sinon.stub().resolves(response),
+    };
+
+    updateStateHandler = updateStateHandlerFactory(
+      rpcClientMock,
+    );
+  });
+
+  afterEach(function afterEach() {
+    this.sinon.restore();
+  });
+
+  it('should throw an error if header is not specified', async () => {
+    call.request.getHeader.returns(null);
+
+    try {
+      await updateStateHandler(call);
+
+      expect.fail('Error was not thrown');
+    } catch (e) {
+      expect(e).to.be.an.instanceOf(InvalidArgumentGrpcError);
+      expect(e.getMessage()).to.equal('Invalid argument: header is not specified');
+    }
+  });
+
+  it('should throw an error if packet is not specified', async () => {
+    call.request.getPacket.returns(null);
+
+    try {
+      await updateStateHandler(call);
+
+      expect.fail('Error was not thrown');
+    } catch (e) {
+      expect(e).to.be.an.instanceOf(InvalidArgumentGrpcError);
+      expect(e.getMessage()).to.equal('Invalid argument: packet is not specified');
+    }
+  });
+
+  it('should return valid result', async () => {
+    const result = await updateStateHandler(call);
+
+    const st = {
+      header: Buffer.from(stHeader).toString('hex'),
+      packet: Buffer.from(stPacket),
+    };
+
+    const tx = cbor.encode(st).toString('base64');
+
+    expect(result).to.be.an.instanceOf(UpdateStateTransitionResponse);
+    expect(rpcClientMock.request).to.be.calledOnceWith('broadcast_tx_commit', { tx });
+  });
+
+  it('should throw InvalidArgumentGrpcError if Tendermint Core returns check_tx with non zero code', async () => {
+    rpcClientMock.request.resolves({
+      check_tx: { code: 1, log },
+      deliver_tx: { log },
+      hash:
+        'B762539A7C17C33A65C46727BFCF2C701390E6AD7DE5190B6CC1CF843CA7E262',
+      height: '24',
+    });
+
+    try {
+      await updateStateHandler(call);
+
+      expect.fail('Error was not thrown');
+    } catch (e) {
+      expect(e).to.be.an.instanceOf(InvalidArgumentGrpcError);
+      expect(e.getMessage()).to.equal('Invalid argument: some message');
+      expect(e.getMetadata()).to.equal('some data');
+    }
+  });
+
+  it('should throw InvalidArgumentGrpcError if Tendermint Core returns deliver_tx with non zero code', async () => {
+    rpcClientMock.request.resolves({
+      check_tx: { log },
+      deliver_tx: { code: 1, log },
+      hash:
+        'B762539A7C17C33A65C46727BFCF2C701390E6AD7DE5190B6CC1CF843CA7E262',
+      height: '24',
+    });
+
+    try {
+      await updateStateHandler(call);
+
+      expect.fail('Error was not thrown');
+    } catch (e) {
+      expect(e).to.be.an.instanceOf(InvalidArgumentGrpcError);
+      expect(e.getMessage()).to.equal('Invalid argument: some message');
+      expect(e.getMetadata()).to.equal('some data');
+    }
+  });
+
+  it('should return error if timeout happened', async () => {
```

Comment:
It it's not implemented yet (pending) just leave it without function body then it will be marked as pending:

```
it('should return an error if timeout happened');
```